ci/deps/pyspark.yml

```diff
@@ -1 +1 @@
-pyspark>=2.4.3
+pyspark>=3.0.0
```

Comment:
do we *need* to change this?

Reply:
I'm not confident this is needed yet. Let me play around with my implementation to see if I can get it to work for pyspark < 3.0.0

(This test currently fails if pyspark < 3.0.0, but it's dependent on my implementation of the `ReductionVectorizedUDF` compilation function)
https://github.com/ibis-project/ibis/blob/7162c6b5156b34851d301afc7831cdb9b96f5e19/ibis/tests/all/test_vectorized_udf.py#L31-L36
)

Reply:
so you can simply xfail *that* test only on this. no need to bump everything (and also not a good idea)

Reply:
The implementation should work now both on pyspark<3.0.0 and pyspark>=3.0.0 so I changed the pin back to `pyspark>=2.4.3`.

There's related issue though: With OmniSciDB and PySpark backends split up in the CI, `pyspark` resolves to `3.0.0`. If `pyspark` is resolving to `3.0.0`, a couple tests need to be tweaked to work (small stuff like `None` vs `np.nan`).

Making these tweaks isn't hard (I had these tweaks in a previous version of the PR (when I was pinning `pyspark>=3.0.0`)  so I can just bring those changes back).

However, **do we want to pin `pyspark` to be < `3.0.0` to ensure compatibility with older versions of `pyspark`**? or is testing using `pyspark` `3.0.0` OK?

Reply:
> However, **do we want to pin `pyspark` to be < `3.0.0` to ensure CI runs with `pyspark` < `3.0.0` and the PySpark backend is compatible with older versions of `pyspark`**? or is testing using `pyspark` `3.0.0` OK?

I'm now pinning `pyspark` to be `<=2.4.6` (equivalent to `<3.0.0`) to make sure the PySpark backend stays compatible with PySpark 2.4